test/unit/plugin/v2/installer_test.rb

```diff
@@ -0,0 +1,414 @@
+    assert_equal expected, @installer.gem_path
+  end
+
+  def test_it_should_know_its_gem_path_with_a_custom_config_dir_from_env
+    ENV['INSPEC_CONFIG_DIR'] = File.join(@config_dir_path, 'empty')
+    expected = File.join(ENV['INSPEC_CONFIG_DIR'], 'gems', @ruby_abi_version)
+    assert_equal expected, @installer.gem_path
+  end
+end
+
+#-----------------------------------------------------------------------#
+# Installing
+#-----------------------------------------------------------------------#
+class PluginInstallerInstallationTests < MiniTest::Test
+  include InstallerTestHelpers
+
+  def test_refuse_to_install_gems_with_wrong_prefix
+    ENV['INSPEC_CONFIG_DIR'] = File.join(@config_dir_path, 'empty')
+    assert_raises(Inspec::Plugin::V2::InstallError) { @installer.install('ordinal_array')}
+  end
+
+  def test_install_a_gem_from_local_file
+    ENV['INSPEC_CONFIG_DIR'] = File.join(@config_dir_path, 'empty')
+    gem_file = File.join(@plugin_fixture_pkg_path, 'inspec-test-fixture-0.1.0.gem')
+    @installer.install('inspec-test-fixture', gem_file: gem_file)
+
+    # Installing a gem places it under the config dir gem area
+    spec_path = File.join(@installer.gem_path, 'specifications', 'inspec-test-fixture-0.1.0.gemspec')
+    assert File.exist?(spec_path), 'After installation from a gem file, the gemspec should be installed to the gem path'
+    installed_gem_base = File.join(@installer.gem_path, 'gems', 'inspec-test-fixture-0.1.0')
+    assert Dir.exist?(installed_gem_base), 'After installation from a gem file, the gem tree should be installed to the gem path'
+
+    # Installation != gem activation
+    spec = Gem::Specification.load(spec_path)
+    refute spec.activated?, 'Installing a gem should not cause the gem to activate'
+  end
+
+  def test_install_a_gem_from_local_file_creates_plugin_json
+    ENV['INSPEC_CONFIG_DIR'] = File.join(@config_dir_path, 'empty')
+    gem_file = File.join(@plugin_fixture_pkg_path, 'inspec-test-fixture-0.1.0.gem')
+    @installer.install('inspec-test-fixture', gem_file: gem_file)
+
+    # Should now be present in plugin.json
+    plugin_json_path = File.join(ENV['INSPEC_CONFIG_DIR'], 'plugins.json')
+    assert File.exist?(plugin_json_path), 'plugins.json should now exist'
+    plugin_json_data = JSON.parse(File.read(plugin_json_path))
+
+    assert_includes plugin_json_data.keys, 'plugins_config_version'
+    assert_equal '1.0.0', plugin_json_data['plugins_config_version'], 'Plugin config version should ve initted to 1.0.0'
+    assert_includes plugin_json_data.keys, 'plugins'
+    assert_kind_of Array, plugin_json_data['plugins']
+    assert_equal 1, plugin_json_data['plugins'].count, 'plugins.json should have one entry'
+    entry = plugin_json_data['plugins'].first
+    assert_kind_of Hash, entry
+    assert_includes entry.keys, 'name'
+    assert_equal 'inspec-test-fixture', entry['name']
+    # TODO: any other fields to check? gem version?
```

Comment:
Should we create/update a issue with a list of these TODO's and maybe title it as a RFC?

Reply:
There is a checklist of minor tech debt here: https://github.com/inspec/inspec/issues/3317